backend/fms_core/serializers.py

```diff
@@ -165,16 +166,62 @@ class ProcessMeasurementExportSerializer(serializers.ModelSerializer):
     source_sample_name = serializers.CharField(read_only=True)
 
     class Meta:
-      model = ProcessMeasurement
-      fields = ('process_measurement_id', 'process_id', 'protocol_name', 'source_sample_name', 'child_sample_name', 'volume_used', 'execution_date', 'comment')
+        model = ProcessMeasurement
+        fields = ('process_measurement_id', 'process_id', 'protocol_name', 'source_sample_name', 'child_sample_name', 'volume_used', 'execution_date', 'comment')
+
+class ProcessMeasurementWithPropertiesExportSerializer(serializers.ModelSerializer):
+    DEFAULT_META_FIELDS = ( 'process_measurement_id',
+                            'process_id',
+                            'protocol_name',
+                            'source_sample_name',
+                            'child_sample_name',
+                            'volume_used',
+                            'execution_date',
+                            'comment' )
+
+    def __init__(self, *args, **kwargs):
+        # Reset Meta fields
+        self.Meta.fields = self.DEFAULT_META_FIELDS
+        # Instantiate the superclass normally
+        super(ProcessMeasurementWithPropertiesExportSerializer, self).__init__(*args, **kwargs)
+        # List all property fields that are tied to the protocol
+        self.property_types = self.list_property_types(self.instance)
+        for property_type in self.property_types:
+            try:
+                self.fields[property_type.name] = serializers.CharField(read_only=True)
+                self.Meta.fields = self.Meta.fields + (property_type.name,)
+            except Exception as err:
+                print(err)
```

Comment:
Maybe there's a better way of handling the error than just printing it

Reply:
yeah this was to debug but i'll replace it with a message to the frontend...